src/compiler/transformers/decorators-to-static/watch-decorator.ts

```diff
@@ -23,22 +15,12 @@ export const watchDecoratorsToStatic = (
   }
 };
 
-const parseWatchDecorator = (
-  config: d.Config,
-  diagnostics: d.Diagnostic[],
-  watchable: Set<string>,
-  method: ts.MethodDeclaration,
-): d.ComponentCompilerWatch[] => {
+const parseWatchDecorator = (method: ts.MethodDeclaration): d.ComponentCompilerWatch[] => {
   const methodName = method.name.getText();
   const decorators = retrieveTsDecorators(method) ?? [];
   return decorators.filter(isDecoratorNamed('Watch')).map((decorator) => {
     const [propName] = getDeclarationParameters<string>(decorator);
-    if (!watchable.has(propName)) {
-      const diagnostic = config.devMode ? buildWarn(diagnostics) : buildError(diagnostics);
-      diagnostic.messageText = `@Watch('${propName}') is trying to watch for changes in a property that does not exist.
-        Make sure only properties decorated with @State() or @Prop() are watched.`;
-      augmentDiagnosticWithNode(diagnostic, decorator);
-    }
```

Comment:
We do lose a compile-time check with this change. Only way around it would be to maintain a list of all the valid HTML attributes to check against which would be a nightmare.

Reply:
In Framework we maintain a list of attributes to inherit here: https://github.com/ionic-team/ionic-framework/blob/f82709595d5031a759e35b11dbcccb0ea5870ce0/core/src/utils/helpers.ts#L118-L175

But our reasoning is to not inherit all attributes that exist on the host. I'm not really sure why you'd restrict what attribute a developer can watch a value change on. That's based on the implementation details of the component, not a compiler constraint.

Reply:
@sean-perkins That's great to know!

Getting that context is what I'm really after here to both:
1. influence my understanding of this particular piece of code
2. add historical context to this PR - this helps prevent the issue where we don't capture this info anywhere in VCS/issue tracking and that information gets lost (which is sadly, lacking in older commits in Stencil/something I'd like us to prevent reverting to)

In that case, I think this code/logic fine as is 👍